site/diff/index.md

```diff
@@ -0,0 +1,425 @@
+# Diff Algorithms
+:summary: How I overcame copying and modifying my own diff library from project to project by \
+          diving too deep into diff algorithms and coming out at the other end with new diff \
+          library in my hand.
+
+For software engineers, diffs are a ubiquitous method for representing changes: We use diffs to
+compare different versions of the same file (e.g., during code review or when trying to understand
+the history of a file), to visualize the difference of a failing test compared with its
+expectation, or to apply changes to source files automatically.
+
+Every project I worked on professionally or privately eventually needed a diff to visualize a change
+or to apply a patch. However, I have never been satisfied with any of the freely available diff
+libraries. This was never really a problem professionally, but for private projects, I have copied
+and modified my own library from project to project until I mentioned this to a colleague who set me
+on the path to publish my Go library (a port of a previous C++ library I used to copy and modify).
+*Boy, did I underestimate how close my library was to publishability!*
+
+Anyway, I did it and I learned a whole lot about diff algorithms. You can find my library at
+[znkr.io/diff](https://znkr.io/diff) and what I learned in this article. I am not finished learning
+yet, so I plan to update this article as my understanding continues to evolve.
+
+## Existing Diff Libraries
+
+Let me start by explaining why I am dissatisfied with existing diff libraries. There are a number of
+attributes that are important to me. Not all of these attributes are important for every use case,
+but a diff library that I can use for all of my use cases needs to fulfil all of them.
```

Comment:
I don’t know if you write American English or British English, but in American English it’d be “fulfill” (two l at the end). I’ll not flag other occurrences, you can use a spell checker to reach consistency one way or the other :)